.github/workflows/gh-page_builder.yml

```diff
@@ -3,6 +3,11 @@ name: JupyterBook
 
 on:
   [push]
+  
+permissions:
+  contents: read
+  pages: write
+  id-token: write
```

Comment:
Do we need the `id-token: write` ? This gives Github Workflow the ability to create a token ID.

Reply:
Yes, `pages: write` is to deploy to Pages, `id-token: write` is to verify the deployment originates from an appropriate source. These two permission grant GITHUB_TOKEN the permissions required to make a Pages deployment.